### PR TITLE
Reinstate integration tests for ncdfgenerator

### PR DESCRIPTION
**Pull request by pmbohm**
This is part of the task of fixing SOOP

### CODE REVIEW

File: src/extension/wps/src/main/java/au/org/emii/ncdfgenerator/NcdfEncoder.java [hunk 1]
```diff
@@ -253,8 +255,13 @@ private Object evaluateSql(long instanceId, String whereClause, String orderClau
...
 
             // TODO more checks around this
-            // maybe support converion to ncdf array attribute types
-            rs.next();
+            // maybe support conversion to ncdf array attribute types
+            if (!rs.next() ) {
+                String err = String.format("No data returned for: \n%s",sql);
+                throw new Exception(err);
+            }
+            //System.out.println(sql);
```
Comment (jonescc) [inline src/extension/wps/src/main/java/au/org/emii/ncdfgenerator/NcdfEncoder.java]: Perhaps remove this - its just noise.

File: src/extension/wps/src/test/java/au/org/emii/ncdfgenerator/GenerationIT.java [hunk 2]
```diff
@@ -179,54 +183,55 @@ private Connection getConn() throws Exception {
         Properties props = new Properties();
         props.setProperty("user", env.get("POSTGRES_USER"));
         props.setProperty("password", env.get("POSTGRES_PASS"));
-        props.setProperty("ssl", "true");
-        props.setProperty("sslfactory", "org.postgresql.ssl.NonValidatingFactory");
+        props.setProperty("database", env.get("POSTGRES_DB"));
+/*        props.setProperty("ssl", "true");
+        props.setProperty("sslfactory", "org.postgresql.ssl.NonValidatingFactory");*/
```
Comment (jonescc) [inline src/extension/wps/src/test/java/au/org/emii/ncdfgenerator/GenerationIT.java]: If we aren't going to use it we should remove it

File: src/extension/wps/src/test/java/au/org/emii/ncdfgenerator/GenerationIT.java [hunk 3]
```diff
@@ -242,42 +247,44 @@ public void setup() throws Exception {
...
+        String cql = "TIME >= '2015-01-13T23:00:00Z'AND TIME <= '2015-01-14T00:00:00Z'";
         MockOutputterCounter outputter = new MockOutputterCounter();
-        NcdfEncoder encoder = getEncoder(config, cql, getConn(), "soop_sst_trajectory", outputter);
+        NcdfEncoder encoder = getEncoder(config, cql,  "soop_sst_trajectory", outputter);
         consumeEncoderOutput(encoder);
     }
 
     @Test
-    public void testAnmnTs() throws Exception {
+      public void testAnmnTs() throws Exception {
```
Comment (jonescc) [inline src/extension/wps/src/test/java/au/org/emii/ncdfgenerator/GenerationIT.java]: Indent problem?

File: src/extension/wps/src/test/resources/soop_sst_trajectory.xml [hunk 4]
```diff
@@ -652,7 +653,7 @@
       </attributes>
     </variable>
     <!-- this is a new variable replacing variable TEMP measured by Fantasea Wonder .The harvester has to be modified to create this new var in the DB -->
-    <!--variable>
+<!--    <variable>
```
Comment (jonescc) [inline src/extension/wps/src/test/resources/soop_sst_trajectory.xml]: Do we need commented out config?

File: src/extension/wps/src/test/resources/soop_sst_trajectory.xml [hunk 5]
```diff
@@ -661,15 +662,15 @@
...
         <attribute name="valid_min" value="-2.5f"/>
         <attribute name="valid_max" value="40.f"/>
         <attribute name="_FillValue" value="999999.f"/>
         <attribute name="coordinates" value="'TIME LATITUDE LONGITUDE'"/>
         <attribute name="ancillary_variables" value="'SST_quality_control'"/>
       </attributes>
-    </variable>
-    <variable>
+    </variable>-->
+ <!--   <variable>
```
Comment (jonescc) [inline src/extension/wps/src/test/resources/soop_sst_trajectory.xml]: And here

**Comment by jonescc** [comment]
@pmbohm looks really good.  Only a few minor comments.

**Comment by jonescc** [comment]
Perhaps should travis/jenkins to use mvn install -DskipITs or mvn package instead of skipping test saltogether (or setup an environment in which to run integration tests).

File: src/extension/wps/src/test/java/au/org/emii/ncdfgenerator/GenerationIT.java (hunk 2 above)
Comment (pmbohm) [inline src/extension/wps/src/test/java/au/org/emii/ncdfgenerator/GenerationIT.java]: I didnt use it but others might

Comment (jonescc) [inline src/extension/wps/src/test/java/au/org/emii/ncdfgenerator/GenerationIT.java]: I reckon we should be setting it up to run one way.  Any reason why we can't connect to the db the same way every time - probably using ssl as thats what we do elsewhere.  No need to cater for manual changes here.

**Comment by pmbohm** [comment]
@jonescc Pull request comments addressed. 
Using Intellij code formating and added a readme.
The tests for SOOP will fail with the current CQL test condition on the current production data. At this stage it looks like the actual SQL in the template will most likely change

File: src/extension/wps/src/main/java/au/org/emii/ncdfgenerator/NcdfEncoder.java [hunk 6]
```diff
@@ -212,38 +214,49 @@ else if (value instanceof Array) {
...
 
         // this is awful...
         // format the file into the output stream
-        InputStream is = createWritable.getStream();
-        outputFormatter.write((String)filename, is);
-        is.close();
+        if (filename != null) {
+            InputStream is = createWritable.getStream();
+            outputFormatter.write((String) filename, is);
+            is.close();
```
Comment (jonescc) [inline src/extension/wps/src/main/java/au/org/emii/ncdfgenerator/NcdfEncoder.java]: I know you didn't write this - but we I just noticed it and we should get into the habit of always ensuring input streams are closed by wrapping in a try resources block .  e.g.


    try (InputStream is = createWritable.getStream()) {
        outputFormatter.write((String) filename, is);
    }

**Comment by pmbohm** [comment]
@jonescc Pull request comments addressed. 
Using Intellij code formatting and added a readme.

File: src/extension/wps/src/main/java/au/org/emii/ncdfgenerator/NcdfEncoder.java [hunk 7]
```diff
@@ -212,38 +212,46 @@ else if (value instanceof Array) {
...
+        Object filename = evaluateSql(replacedSql);
 
-        // this is awful...
-        // format the file into the output stream
-        InputStream is = createWritable.getStream();
-        outputFormatter.write((String)filename, is);
-        is.close();
+        try (InputStream is = createWritable.getStream()) {
+            outputFormatter.write((String) filename, is);
+            is.close();
```
Comment (jonescc) [inline src/extension/wps/src/main/java/au/org/emii/ncdfgenerator/NcdfEncoder.java]: Bugger - just noticed this line is still here.  Its now done implicitly by the try when exiting the try block and is not required (although it won't cause any issues).   I'll remove in my changes

**Pull request merged**